sdk/storage/Azure.Storage.Blobs/src/BlobServiceClient.cs

```diff
@@ -1749,5 +1773,189 @@ internal BlobServiceClient(Uri serviceUri, HttpPipelinePolicy authentication, Bl
             }
         }
         #endregion FilterBlobs
+
+        #region GenerateSAS
+        /// <summary>
+        /// The <see cref="GetSasBuilder"/> returns a <see cref="BlobSasBuilder"/> that
+        /// sets the respective properties in the BlobSasBuilder from the client.
+        ///
+        /// Note that properties in the returned builder will not set the BlobName
+        /// </summary>
+        /// <param name="permissions">
+        /// Specifies the list of permissions that can be set in the returned SasBuilder
+        /// See <see cref="BlobSasPermissions"/>.
+        /// </param>
+        /// <param name="expiresOn">
+        /// Specifies when to set the expires time in the returned Sas builder.
+        /// </param>
+        /// <returns>
+        /// A <see cref="BlobSasBuilder"/> on successfully deleting.
+        /// </returns>
+        /// <remarks>
+        /// A <see cref="RequestFailedException"/> will be thrown if
+        /// a failure occurs.
+        /// </remarks>
+        public BlobSasBuilder GetSasBuilder(
```

Comment:
I think the service client should just have the ability to create account sas.

Reply:
What about service SAS?

Reply:
The service client is an account level client so it doesnt make sense to make a service sas on this client

Reply:
I think it'd be interesting to consider a `BlobServiceClient.GetAccountSasBuilder` method in the future after we see how customers react to these APIs.  We'd have to add a similar `GenerateAccountSasUri(BlobAccountSasBuilder builder)` method too.